src/devtools/client/webconsole/components/Output/ConsoleOutput.tsx

```diff
@@ -88,12 +89,22 @@ class ConsoleOutput extends React.Component<PropsFromRedux> {
   }
 
   maybeScrollToResult(prevProps: PropsFromRedux) {
-    const messagesDelta = this.props.messages.size - prevProps.messages.size;
+    const messagesDelta = this.props.messagesCount - prevProps.messagesCount;
+
+    const { entities, ids } = this.props.messages;
+    if (ids.length === 0) {
+      return;
+    }
+
+    // We're not sorting the IDs array, so should be latest added
+    const lastMessageId = ids[ids.length - 1];
+    const lastMessage = entities[lastMessageId]!;
 
     // [...this.props.messages.values()] seems slow
     // we should have a separate messageList somewhere we can check OR
     // use a memoization function to be able to get the last message quickly
-    const lastMessage = [...this.props.messages.values()][this.props.messages.size - 1];
+    // const lastMessage = [...this.props.messages.values()][this.props.messages.size - 1];
```

Comment:
Can we delete this?

Reply:
Ah yes, that should go away.  I'll take care of that in the follow-up PR that I'm already working on.